src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -1535,6 +1535,18 @@ public function testEnvCaseInsensitiveOnWindows()
         }
     }
 
+    public function testProcessWaitingCallback()
+    {
+        $p = $this->getProcessForCode('usleep(3000000);');
+        $callCounter = 0;
+        $p->wait(null,
+            function ($type, $buffer) use (&$callCounter) {
```

Comment:
Why is `$type` and `$buffer`? Your implementation passes the actual `Process` instance to the callback.